cli/crates/cli/src/cli_input.rs

```diff
@@ -1,11 +1,42 @@
 use crate::create::CreateArguments;
 use clap::{arg, command, CommandFactory, Parser, ValueEnum};
 use clap_complete::{shells, Generator};
-use common::consts::{DEFAULT_LOG_FILTER, TRACE_LOG_FILTER};
+use common::{
+    consts::{DEFAULT_LOG_FILTER, TRACE_LOG_FILTER},
+    types::LogLevel,
+};
 use std::{fmt, path::PathBuf};
 
 const DEFAULT_PORT: u16 = 4000;
 
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd, serde::Deserialize, clap::ValueEnum)]
+#[clap(rename_all = "snake_case")]
+pub enum LogLevelFilter {
+    None,
+    Error,
+    Warn,
+    Info,
+    Debug,
+}
+
+impl From<LogLevelFilter> for Option<LogLevel> {
+    fn from(value: LogLevelFilter) -> Self {
```

Comment:
Not sure how that would work out, but I wonder if this could just be done with something like `impl PartialOrd<LogLevel> for LogLevelFilter` or something like that

Reply:
I considered that but to me operations like `PartialOrd` etc. should be defined between types that are equivalent. Semantically, these aren't, actually. `LogLevel` is the level of a single message, whereas filter belongs to another category (includes `None` that a level can't be). Doesn't feel like `PartialOrd` would be appropriate…

Reply:
Well don't have a strong opinion, I think it probably would have made code simpler rather than dealing with the optional special case. 🤷